httpbp/response.go

```diff
@@ -0,0 +1,135 @@
+package httpbp
+
+import (
+	"context"
+	"encoding/json"
+	"html/template"
+	"net/http"
+
+	httpgk "github.com/go-kit/kit/transport/http"
+)
+
+// ResponseCookies is an interface that your Response objects can implement in
+// order to have the httpbp.Encode methods automatically add cookies to the
+// response.
+type ResponseCookies interface {
+	// Return a list of all cookies to set on the response.
+	Cookies() []*http.Cookie
+
+	// Add a new cookie to set on the response.
+	AddCookie(*http.Cookie)
+}
+
+// ResponseHeaders is an interface that your Response objects can implement in
+// order to have the httpbp.Encode methods automatically add http headers to the
+// response.
+type ResponseHeaders interface {
+	// Headers returns the http.Header container that can be used to add headers
+	// to the response.
+	Headers() http.Header
+}
+
+// ErrorResponse is an interface that your Response objects can implement in
+// order to have the httpbp.Encode methods automatically return http errors.
+type ErrorResponse interface {
+	// Err returns the error set on the response.
+	//
+	// This method is named "Err" rather than "Error" to avoid confusion with
+	// the "Error" function that is specified in the error interface.
+	Err() error
+
+	// SetError sets an error to be returned in the response.
+	SetError(err error)
```

Comment:
is this function ever used?

Reply:
I'm not sure it's necessary... I was thinking it would be good to have to clarify "this is the thing you use to set the error returned by `Err`"

Reply:
i think most implementations will just change the underlying field directly, without calling a setter.

Reply:
Here we are defining an interface for callers to implement and for us to call, the principle to that is to only define the functions we need to call. `SetErr` is not something we'll call, even if it's useful to the user/implementation.